### DISCUSSION TITLE
Default implementing GetEnumerator() in foreach loop for Range

### DISCUSSION THREAD

**Discussion by DanielNikoofar** [dotnet/csharplang#6207] [Language Ideas] [answered]
Hello,

If developers want to use Range in foreach loop, they see Compiler Error CS1579. So they must implement GetEnumerator().
Please implementing GetEnumerator() in default.

**Comment by PathogenDavid** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/discussions/3004 and more specifically https://github.com/dotnet/runtime/issues/462.
